mesonbuild/environment.py

```diff
@@ -1378,32 +1379,45 @@ def detect_rust_compiler(self, for_machine):
                 # the default use that, and second add the necessary arguments
                 # to rust to use -fuse-ld
 
-                extra_args = {}
-                always_args = []
-                if is_link_exe:
-                    compiler.extend(['-C', 'linker={}'.format(cc.linker.exelist[0])])
-                    extra_args['direct'] = True
-                    extra_args['machine'] = cc.linker.machine
-                elif not ((info.is_darwin() and isinstance(cc, AppleClangCCompiler)) or
-                          isinstance(cc, GnuCCompiler)):
+                if override is None:
+                    extra_args = {}
+                    always_args = []
+                    if is_link_exe:
+                        compiler.extend(['-C', 'linker={}'.format(cc.linker.exelist[0])])
+                        extra_args['direct'] = True
+                        extra_args['machine'] = cc.linker.machine
+                    elif not ((info.is_darwin() and isinstance(cc, AppleClangCCompiler)) or
+                            isinstance(cc, GnuCCompiler)):
```

Comment:
__[Flake8]__ continuation line under-indented for visual indent ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6457#issue-5057266))

<details>

| Rule |
| - |
| E128 |

</details>
<sub>posted by Sider</sub>